### ISSUE TITLE
Setup:upgrade fails with the error message: Smile\ElasticsuiteCore\Helper\IndexSettings::getIndexAlias(): Return value must be of type string, null returned

### ISSUE THREAD

**Issue by machrafemergento** [Smile-SA/elasticsuite#2716]
Installation fails at `setup:upgrade` with the following error message:
```
[159.223.7.29] Smile\ElasticsuiteCore\Helper\IndexSettings::getIndexAlias(): Return value must be of type string, null returned
[159.223.7.29] #0 /vendor/smile/elasticsuite/src/module-elasticsuite-core/Helper/IndexSettings.php(148): Smile\ElasticsuiteCore\Helper\IndexSettings->getIndexAlias()
[159.223.7.29] #1 /vendor/smile/elasticsuite/src/module-elasticsuite-core/Index/IndexSettings.php(106): Smile\ElasticsuiteCore\Helper\IndexSettings->getIndexAliasFromIdentifier()
[159.223.7.29] #2 /vendor/smile/elasticsuite/src/module-elasticsuite-core/Index/IndexOperation.php(349): Smile\ElasticsuiteCore\Index\IndexSettings->getIndexAliasFromIdentifier()
[159.223.7.29] #3 /vendor/smile/elasticsuite/src/module-elasticsuite-core/Index/IndexOperation.php(133): Smile\ElasticsuiteCore\Index\IndexOperation->initIndex()
[159.223.7.29] #4 /vendor/smile/elasticsuite/src/module-elasticsuite-core/Indexer/GenericIndexerHandler.php(169): Smile\ElasticsuiteCore\Index\IndexOperation->createIndex()
[159.223.7.29] #5 /vendor/smile/elasticsuite/src/module-elasticsuite-catalog/Model/Category/Indexer/Fulltext.php(102): Smile\ElasticsuiteCore\Indexer\GenericIndexerHandler->cleanIndex()
[159.223.7.29] #6 /vendor/magento/framework/Interception/Interceptor.php(58): Smile\ElasticsuiteCatalog\Model\Category\Indexer\Fulltext->executeFull()
[159.223.7.29] #7 /vendor/magento/framework/Interception/Interceptor.php(138): Smile\ElasticsuiteCatalog\Model\Category\Indexer\Fulltext\Interceptor->___callParent()
[159.223.7.29] #8 /vendor/magento/framework/Interception/Interceptor.php(153): Smile\ElasticsuiteCatalog\Model\Category\Indexer\Fulltext\Interceptor->Magento\Framework\Interception\{closure}()
[159.223.7.29] #9 /generated/code/Smile/ElasticsuiteCatalog/Model/Category/Indexer/Fulltext/Interceptor.php(23): Smile\ElasticsuiteCatalog\Model\Category\Indexer\Fulltext\Interceptor->___callPlugins()
[159.223.7.29] #10 /vendor/magento/module-indexer/Model/Indexer.php(443): Smile\ElasticsuiteCatalog\Model\Category\Indexer\Fulltext\Interceptor->executeFull()
[159.223.7.29] #11 /vendor/magento/module-indexer/Model/Indexer/DependencyDecorator.php(268): Magento\Indexer\Model\Indexer->reindexAll()
[159.223.7.29] #12 /vendor/smile/elasticsuite/src/module-elasticsuite-catalog/Setup/InstallData.php(96): Magento\Indexer\Model\Indexer\DependencyDecorator->reindexAll()
[159.223.7.29] #13 /setup/src/Magento/Setup/Model/Installer.php(1072): Smile\ElasticsuiteCatalog\Setup\InstallData->install()
[159.223.7.29] #14 /setup/src/Magento/Setup/Model/Installer.php(961): Magento\Setup\Model\Installer->handleDBSchemaData()
[159.223.7.29] #15 /setup/src/Magento/Setup/Console/Command/UpgradeCommand.php(147): Magento\Setup\Model\Installer->installDataFixtures()
[159.223.7.29] #16 /vendor/symfony/console/Command/Command.php(255): Magento\Setup\Console\Command\UpgradeCommand->execute()
[159.223.7.29] #17 /vendor/symfony/console/Application.php(1009): Symfony\Component\Console\Command\Command->run()
[159.223.7.29] #18 /vendor/symfony/console/Application.php(273): Symfony\Component\Console\Application->doRunCommand()
[159.223.7.29] #19 /vendor/magento/framework/Console/Cli.php(116): Symfony\Component\Console\Application->doRun()
[159.223.7.29] #20 /vendor/symfony/console/Application.php(149): Magento\Framework\Console\Cli->doRun()
[159.223.7.29] #21 /bin/magento(23): Symfony\Component\Console\Application->run()
```

**Additional context**
- Magento version is 2.5.5
- ElastcSuite version: I downloaded the 2.10.x branch, I don't know which commit though.
- ElasticSearch version is 7.17.3
- The `setup:upgrade` command is run by the root user.
- This error is only happening in the Stage/Prod env, ElasticSuite works just fine on my dev env.

What could be the problem?

**Comment by vahonc** [comment]
Hello @machrafemergento,

First of all, check if ElasticSuite is selected as the default **Search Engine** as it showed on the screenshot below:

![Screenshot from 2022-09-20 10-26-38](https://user-images.githubusercontent.com/67791361/191194473-06547a04-77bb-4b94-a932-cdff9ca9182e.png)

Also, please read our [Wiki](https://github.com/Smile-SA/elasticsuite/wiki/ModuleInstall#indices-settings) and check your Indices configuration.

BR,
Vadym

**Comment by machrafemergento** [comment]
Hi @vahonc, Thanks for the reply.

I no longer have this error, my bad.
To give some context, I was getting this error in a deployment pipeline so I tried running `setup:upgrade` manually and somehow it worked, I ran the pipeline again to make sure and it surprisingly worked without issues.
I apologize, I really don't know what's happened, but it's working fine now.

Feel free to close or delete the issue.
